### PR TITLE
Repositories

### PR DESCRIPTION
**Pull request by dbogdan2000**
Created TaskRepository, UserTaskRepository, VTaskRepository and VUserTaskRepository. Implemented methods Create, Update and Delete in Repository. Added method DeleteUserProcedure to UserProfileRepository and method DeleteTaskProcedure to TaskRepository. This methods allow you to remove user or task by calling appropriate procedure.

### CODE REVIEW

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/UserProfileRepository.cs [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+        private readonly DIMSCoreContext _context;
         public UserProfileRepository(DIMSCoreContext context) : base(context)
         {
+            _context = context;
+        }
+
+        public async ThreadTask DeleteUserProcedure(int id)
+        {
+            await _context.Database.ExecuteSqlRawAsync("Exec DeleteUser @UserId", id);
         }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/UserProfileRepository.cs:20]: Again about virtual keyword

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/VTaskRepository.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        
+        public VTaskRepository(DIMSCoreContext context)
+        {
+            _context = context;
+        }
+
+        public IQueryable<VTask> GetAll()
+        {
+            return _context.VTasks.AsNoTracking();
+        }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/VTaskRepository.cs:20]: Don't duplicate logic. Look at Repository class and make up similar class for readonly repository.

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi dude!
I did some comment. I hope you will figure them out. If not just let me know and I'll explain whole.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskRepository.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace DIMS_Core.DataAccessLayer.Repositories
+{
+    public class TaskRepository : Repository<Task>
+    {
+        private readonly DIMSCoreContext _context;
+        
+        public TaskRepository(DIMSCoreContext context) : base(context)
+        {
+            _context = context;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskRepository.cs:14]: Repository already has the same object. If you need to get some prop from this object you have to create appropriate method.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/UserProfileRepository.cs [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
+using ThreadTask = System.Threading.Tasks.Task;
 
 namespace DIMS_Core.DataAccessLayer.Repositories
 {
     public class UserProfileRepository : Repository<UserProfile>
     {
+        private readonly DIMSCoreContext _context;
         public UserProfileRepository(DIMSCoreContext context) : base(context)
         {
+            _context = context;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/UserProfileRepository.cs:14]: Again don't duplicate objects

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
I did some comments. Let me know if any questions.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/Repository.cs [hunk 5]
```diff
@@ -58,17 +59,25 @@ public Task<TEntity> GetById(int id)
...
 
-        public Task Delete(int id)
+        public virtual async Task Delete(int id)
         {
-            throw new NotImplementedException();
+            var entity = await _set.FindAsync(id);
+            _set.Remove(entity);
+        }
+
+        protected DatabaseFacade GetDb()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/Repository.cs:78]: Bad name. I don't understand what is doing this method so I have to look it through. I think GetDatabaseFacade will be better.

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi man.
Need one small change

**Review by PushMeNow — APPROVED** [review]
Approved

**Review by STALINovsky — APPROVED** [review]
Looks nice, approved.

**Pull request merged**